### PR TITLE
3.0.0-alpha.4

### PR DESCRIPTION
**Pull request by nickpatrick**
- Add user segments and top chains
- Add `getLocationWithDesiredAccuracy:completionHandler:`

### CODE REVIEW

File: RadarSDK/RadarSegment.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ Represents a user segment.
+ */
+@interface RadarSegment : NSObject
+
+/**
+ The description of the segment.
+ */
+@property (nonnull, copy, nonatomic, readonly) NSString *_description;
```
Comment (jsani-radar) [inline RadarSDK/RadarSegment.h]: nit: is this a private variable due to the `_`? I'd assume `description` is meant to be available to clients

File: RadarSDK/RadarSegment.m [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+#import "RadarSegment+Internal.h"
+
+@implementation RadarSegment
+
+- (instancetype)initWithDescription:(nonnull NSString *)description
+                         externalId:(nonnull NSString *)externalId {
+    self = [super init];
+    if (self) {
+        __description = description;
```
Comment (jsani-radar) [inline RadarSDK/RadarSegment.m]: ditto: private var

File: RadarSDK/RadarUser.m [hunk 3]
```diff
@@ -184,11 +189,41 @@ - (instancetype _Nullable)initWithObject:(NSObject *)object {
...
+            RadarSegment *segment = [[RadarSegment alloc] initWithObject:segmentObj];
+            if (segment) {
+                [mutableSegments addObject:segment];
+            }
+        }
+        
+        userSegments = mutableSegments;
+    }
+    
+    id userTopChainsObj = userDict[@"topChains"];
```
Comment (jsani-radar) [inline RadarSDK/RadarUser.m]: for consistency, can we put the array deserializers in the class -- https://github.com/radarlabs/radar-sdk-android/blob/c661f33d6fd9482c9e47c57a1746ec8cd27f2649/sdk/src/main/java/io/radar/sdk/model/RadarChain.kt#L50-L58

File: RadarSDK/RadarSegment.h (hunk 1 above)
Comment (nickpatrick) [reply RadarSDK/RadarSegment.h]: It conflicts with `description` property on `NSObject` (https://developer.apple.com/documentation/objectivec/nsobjectprotocol/1418746-description). If I could go back in time, I would call this field `name`. Alas...

File: RadarSDK/RadarSegment.m (hunk 2 above)
Comment (nickpatrick) [reply RadarSDK/RadarSegment.m]: See above

File: RadarSDK/RadarUser.m (hunk 3 above)
Comment (nickpatrick) [reply RadarSDK/RadarUser.m]: I'm going to come back and do a final cleanup pass on the deserializers and the test API responses, but will do that in a separate PR

**Comment by nickpatrick** [comment]
@jsani-radar Thanks so much for taking a look at this late last night. I replied to your comments

File: RadarSDK/RadarSegment+Internal.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+//
+//  Copyright © 2019 Radar Labs, Inc. All rights reserved.
+//
+
+#import "RadarSegment.h"
+
+@interface RadarSegment ()
+
+- (nonnull instancetype)initWithDescription:(nonnull NSString *)description
+                                 externalId:(nonnull NSString *)description;
```
Comment (jsani-radar) [inline RadarSDK/RadarSegment+Internal.h]: nit: `externalId` for inline param name

Comment (nickpatrick) [reply RadarSDK/RadarSegment+Internal.h]: Good catch, fixing

Comment (nickpatrick) [reply RadarSDK/RadarSegment+Internal.h]: @jsani-radar Done. Thanks!

Comment (jsani-radar) [reply RadarSDK/RadarSegment+Internal.h]: approved

**Pull request merged**